### PR TITLE
[FIX] fix binaryPredicte's equals function ignore op

### LINKED ISSUE #3756: BinaryPredict's equals function compares ignore the op
**Describe the bug**
BinaryPredict's equals function compares ignore the op, the  opcode is not init in this class, it will always be INVALID_OPCODE, so if a BinaryPredict has same child  it will be qual, like `a<1` and `a>=1`
```
    @Override
    public boolean equals(Object obj) {
        if (!super.equals(obj)) {
            return false;
        }
        return ((BinaryPredicate) obj).opcode == this.opcode;
    }
```

### PR DESCRIPTION
**Pull request by yangzhg**
BinaryPredicte's equals funtion compare by opccode ,
but the opcode is never inited, 
so it will return true if this child is same,  for example `a>1` and `a<1` are equal
Fixes #3756

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java [hunk 1]
```diff
@@ -225,7 +226,7 @@ public boolean equals(Object obj) {
         if (!super.equals(obj)) {
             return false;
         }
-        return ((BinaryPredicate) obj).opcode == this.opcode;
+        return ((BinaryPredicate) obj).opcode == this.opcode && op.equals(((BinaryPredicate) obj).getOp());
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java:229]: I think only compare op is enough.

Comment (morningman) [reply fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java:229]: It only for safe, don't know what the `opcode` for.
If it is useless, we can remove it in next PR.

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**